File: binding/src/libqgraphicslinearlayout.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qgraphicslinearlayout.html
pub const qgraphicslinearlayout = struct {

    /// New constructs a new QGraphicsLinearLayout object.
    pub fn New() QtC.QGraphicsLinearLayout {
        return qtc.QGraphicsLinearLayout_new();
    }


    /// New2 constructs a new QGraphicsLinearLayout object.
    pub fn New2(orientation: i32) QtC.QGraphicsLinearLayout {
        return qtc.QGraphicsLinearLayout_new2(@intCast(orientation));
    }


    /// New3 constructs a new QGraphicsLinearLayout object.
    pub fn New3(parent: ?*anyopaque) QtC.QGraphicsLinearLayout {
        return qtc.QGraphicsLinearLayout_new3(@ptrCast(parent));
    }


    /// New4 constructs a new QGraphicsLinearLayout object.
    pub fn New4(orientation: i32, parent: ?*anyopaque) QtC.QGraphicsLinearLayout {
        return qtc.QGraphicsLinearLayout_new4(@intCast(orientation), @ptrCast(parent));
    }


    pub fn SetOrientation(self: ?*anyopaque, orientation: i32) void {
        qtc.QGraphicsLinearLayout_SetOrientation(@ptrCast(self), @intCast(orientation));
    }

    pub fn Orientation(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsLinearLayout_Orientation(@ptrCast(self));
    }

    pub fn AddItem(self: ?*anyopaque, item: ?*anyopaque) void {
        qtc.QGraphicsLinearLayout_AddItem(@ptrCast(self), @ptrCast(item));
    }

    pub fn AddStretch(self: ?*anyopaque, ) void {
        qtc.QGraphicsLinearLayout_AddStretch(@ptrCast(self));
    }

    pub fn InsertItem(self: ?*anyopaque, index: i32, item: ?*anyopaque) void {
        qtc.QGraphicsLinearLayout_InsertItem(@ptrCast(self), index, @ptrCast(item));
    }

    pub fn InsertStretch(self: ?*anyopaque, index: i32) void {
        qtc.QGraphicsLinearLayout_InsertStretch(@ptrCast(self), index);
    }

    pub fn RemoveItem(self: ?*anyopaque, item: ?*anyopaque) void {
        qtc.QGraphicsLinearLayout_RemoveItem(@ptrCast(self), @ptrCast(item));
    }

    pub fn RemoveAt(self: ?*anyopaque, index: i32) void {
        qtc.QGraphicsLinearLayout_RemoveAt(@ptrCast(self), index);
    }

    /// Allows for overriding the related default method
    pub fn OnRemoveAt(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QGraphicsLinearLayout_OnRemoveAt(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseRemoveAt(self: ?*anyopaque, index: i32) void {
        qtc.QGraphicsLinearLayout_QBaseRemoveAt(@ptrCast(self), index);
    }

    pub fn SetSpacing(self: ?*anyopaque, spacing: f64) void {
        qtc.QGraphicsLinearLayout_SetSpacing(@ptrCast(self), spacing);
    }

    pub fn Spacing(self: ?*anyopaque, ) f64 {
        return qtc.QGraphicsLinearLayout_Spacing(@ptrCast(self));
    }

    pub fn SetItemSpacing(self: ?*anyopaque, index: i32, spacing: f64) void {
        qtc.QGraphicsLinearLayout_SetItemSpacing(@ptrCast(self), index, spacing);
    }

    pub fn ItemSpacing(self: ?*anyopaque, index: i32) f64 {
        return qtc.QGraphicsLinearLayout_ItemSpacing(@ptrCast(self), index);
    }

    pub fn SetStretchFactor(self: ?*anyopaque, item: ?*anyopaque, stretch: i32) void {
        qtc.QGraphicsLinearLayout_SetStretchFactor(@ptrCast(self), @ptrCast(item), stretch);
    }

    pub fn StretchFactor(self: ?*anyopaque, item: ?*anyopaque) i32 {
        return qtc.QGraphicsLinearLayout_StretchFactor(@ptrCast(self), @ptrCast(item));
    }

    pub fn SetAlignment(self: ?*anyopaque, item: ?*anyopaque, alignment: i32) void {
        qtc.QGraphicsLinearLayout_SetAlignment(@ptrCast(self), @ptrCast(item), @intCast(alignment));
    }

    pub fn Alignment(self: ?*anyopaque, item: ?*anyopaque) i32 {
        return qtc.QGraphicsLinearLayout_Alignment(@ptrCast(self), @ptrCast(item));
    }

    pub fn SetGeometry(self: ?*anyopaque, rect: ?*anyopaque) void {
        qtc.QGraphicsLinearLayout_SetGeometry(@ptrCast(self), @ptrCast(rect));
    }

    /// Allows for overriding the related default method
    pub fn OnSetGeometry(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsLinearLayout_OnSetGeometry(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetGeometry(self: ?*anyopaque, rect: ?*anyopaque) void {
        qtc.QGraphicsLinearLayout_QBaseSetGeometry(@ptrCast(self), @ptrCast(rect));
    }

    pub fn Count(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsLinearLayout_Count(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnCount(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QGraphicsLinearLayout_OnCount(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCount(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsLinearLayout_QBaseCount(@ptrCast(self));
    }

    pub fn ItemAt(self: ?*anyopaque, index: i32) QtC.QGraphicsLayoutItem {
        return qtc.QGraphicsLinearLayout_ItemAt(@ptrCast(self), index);
    }

    /// Allows for overriding the related default method
    pub fn OnItemAt(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) QtC.QGraphicsLayoutItem) void {
        qtc.QGraphicsLinearLayout_OnItemAt(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseItemAt(self: ?*anyopaque, index: i32) QtC.QGraphicsLayoutItem {
        return qtc.QGraphicsLinearLayout_QBaseItemAt(@ptrCast(self), index);
    }

    pub fn Invalidate(self: ?*anyopaque, ) void {
        qtc.QGraphicsLinearLayout_Invalidate(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnInvalidate(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsLinearLayout_OnInvalidate(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInvalidate(self: ?*anyopaque, ) void {
        qtc.QGraphicsLinearLayout_QBaseInvalidate(@ptrCast(self));
    }

    pub fn SizeHint(self: ?*anyopaque, which: i32, constraint: ?*anyopaque) QtC.QSizeF {
        return qtc.QGraphicsLinearLayout_SizeHint(@ptrCast(self), @intCast(which), @ptrCast(constraint));
    }

    /// Allows for overriding the related default method
    pub fn OnSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, ?*anyopaque) callconv(.c) QtC.QSizeF) void {
        qtc.QGraphicsLinearLayout_OnSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSizeHint(self: ?*anyopaque, which: i32, constraint: ?*anyopaque) QtC.QSizeF {
        return qtc.QGraphicsLinearLayout_QBaseSizeHint(@ptrCast(self), @intCast(which), @ptrCast(constraint));
    }

    pub fn Dump(self: ?*anyopaque, ) void {
        qtc.QGraphicsLinearLayout_Dump(@ptrCast(self));
    }

    pub fn AddStretch1(self: ?*anyopaque, stretch: i32) void {
        qtc.QGraphicsLinearLayout_AddStretch1(@ptrCast(self), stretch);
    }

    pub fn InsertStretch2(self: ?*anyopaque, index: i32, stretch: i32) void {
        qtc.QGraphicsLinearLayout_InsertStretch2(@ptrCast(self), index, stretch);
    }

    pub fn Dump1(self: ?*anyopaque, indent: i32) void {
        qtc.QGraphicsLinearLayout_Dump1(@ptrCast(self), indent);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QGraphicsLinearLayout_Delete(@ptrCast(self));
    }
};

File: binding/src/libqgraphicslinearlayout.cpp
#include <QGraphicsLayout>
#include <QGraphicsLayoutItem>
#include <QGraphicsLinearLayout>
#include <QRectF>
#include <QSizeF>
#include <qgraphicslinearlayout.h>
#include "libqgraphicslinearlayout.h"
#include "libqgraphicslinearlayout.hxx"

QGraphicsLinearLayout* QGraphicsLinearLayout_new() {
	 return new VirtualQGraphicsLinearLayout();
}

QGraphicsLinearLayout* QGraphicsLinearLayout_new2(int orientation) {
	 return new VirtualQGraphicsLinearLayout(static_cast<Qt::Orientation>(orientation));
}

QGraphicsLinearLayout* QGraphicsLinearLayout_new3(QGraphicsLayoutItem* parent) {
	 return new VirtualQGraphicsLinearLayout(parent);
}

QGraphicsLinearLayout* QGraphicsLinearLayout_new4(int orientation, QGraphicsLayoutItem* parent) {
	 return new VirtualQGraphicsLinearLayout(static_cast<Qt::Orientation>(orientation), parent);
}

void QGraphicsLinearLayout_SetOrientation(QGraphicsLinearLayout* self, int orientation) {
	self->setOrientation(static_cast<Qt::Orientation>(orientation));
}

int QGraphicsLinearLayout_Orientation(const QGraphicsLinearLayout* self) {
	return self->orientation();
}

void QGraphicsLinearLayout_AddItem(QGraphicsLinearLayout* self, QGraphicsLayoutItem* item) {
	self->addItem(item);
}

void QGraphicsLinearLayout_AddStretch(QGraphicsLinearLayout* self) {
	self->addStretch();
}

void QGraphicsLinearLayout_InsertItem(QGraphicsLinearLayout* self, int index, QGraphicsLayoutItem* item) {
	self->insertItem(index, item);
}

void QGraphicsLinearLayout_InsertStretch(QGraphicsLinearLayout* self, int index) {
	self->insertStretch(index);
}

void QGraphicsLinearLayout_RemoveItem(QGraphicsLinearLayout* self, QGraphicsLayoutItem* item) {
	self->removeItem(item);
}

void QGraphicsLinearLayout_RemoveAt(QGraphicsLinearLayout* self, int index) {
	self->removeAt(index);
}

void QGraphicsLinearLayout_SetSpacing(QGraphicsLinearLayout* self, double spacing) {
	self->setSpacing(spacing);
}

double QGraphicsLinearLayout_Spacing(const QGraphicsLinearLayout* self) {
	return self->spacing();
}

void QGraphicsLinearLayout_SetItemSpacing(QGraphicsLinearLayout* self, int index, double spacing) {
	self->setItemSpacing(index, spacing);
}

double QGraphicsLinearLayout_ItemSpacing(const QGraphicsLinearLayout* self, int index) {
	return self->itemSpacing(index);
}

void QGraphicsLinearLayout_SetStretchFactor(QGraphicsLinearLayout* self, QGraphicsLayoutItem* item, int stretch) {
	self->setStretchFactor(item, stretch);
}

int QGraphicsLinearLayout_StretchFactor(const QGraphicsLinearLayout* self, QGraphicsLayoutItem* item) {
	return self->stretchFactor(item);
}

void QGraphicsLinearLayout_SetAlignment(QGraphicsLinearLayout* self, QGraphicsLayoutItem* item, int alignment) {
	self->setAlignment(item, static_cast<QFlags<Qt::AlignmentFlag>>(alignment));
}

int QGraphicsLinearLayout_Alignment(const QGraphicsLinearLayout* self, QGraphicsLayoutItem* item) {
	return self->alignment(item);
}

void QGraphicsLinearLayout_SetGeometry(QGraphicsLinearLayout* self, const QRectF* rect) {
	self->setGeometry(*rect);
}

int QGraphicsLinearLayout_Count(const QGraphicsLinearLayout* self) {
	return self->count();
}

QGraphicsLayoutItem* QGraphicsLinearLayout_ItemAt(const QGraphicsLinearLayout* self, int index) {
	return self->itemAt(index);
}

void QGraphicsLinearLayout_Invalidate(QGraphicsLinearLayout* self) {
	self->invalidate();
}

QSizeF* QGraphicsLinearLayout_SizeHint(const QGraphicsLinearLayout* self, int which, const QSizeF* constraint) {
	return new QSizeF(self->sizeHint(static_cast<Qt::SizeHint>(which), *constraint));
}

void QGraphicsLinearLayout_Dump(const QGraphicsLinearLayout* self) {
	self->dump();
}

void QGraphicsLinearLayout_AddStretch1(QGraphicsLinearLayout* self, int stretch) {
	self->addStretch(stretch);
}

void QGraphicsLinearLayout_InsertStretch2(QGraphicsLinearLayout* self, int index, int stretch) {
	self->insertStretch(index, stretch);
}

void QGraphicsLinearLayout_Dump1(const QGraphicsLinearLayout* self, int indent) {
	self->dump(indent);
}

// Base class handler implementation
void QGraphicsLinearLayout_QBaseRemoveAt(QGraphicsLinearLayout* self, int index) {
	auto* vqgraphicslinearlayout = dynamic_cast<VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_RemoveAt_IsBase(true);
	vqgraphicslinearlayout->removeAt(index);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLinearLayout_OnRemoveAt(QGraphicsLinearLayout* self, intptr_t slot) {
	auto* vqgraphicslinearlayout = dynamic_cast<VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_RemoveAt_Callback(reinterpret_cast<VirtualQGraphicsLinearLayout::QGraphicsLinearLayout_RemoveAt_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsLinearLayout_QBaseSetGeometry(QGraphicsLinearLayout* self, const QRectF* rect) {
	auto* vqgraphicslinearlayout = dynamic_cast<VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_SetGeometry_IsBase(true);
	vqgraphicslinearlayout->setGeometry(*rect);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLinearLayout_OnSetGeometry(QGraphicsLinearLayout* self, intptr_t slot) {
	auto* vqgraphicslinearlayout = dynamic_cast<VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_SetGeometry_Callback(reinterpret_cast<VirtualQGraphicsLinearLayout::QGraphicsLinearLayout_SetGeometry_Callback>(slot));
}
}

// Base class handler implementation
int QGraphicsLinearLayout_QBaseCount(const QGraphicsLinearLayout* self) {
	auto* vqgraphicslinearlayout = dynamic_cast<const VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_Count_IsBase(true);
	return vqgraphicslinearlayout->count();
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLinearLayout_OnCount(const QGraphicsLinearLayout* self, intptr_t slot) {
	auto* vqgraphicslinearlayout = dynamic_cast<const VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_Count_Callback(reinterpret_cast<VirtualQGraphicsLinearLayout::QGraphicsLinearLayout_Count_Callback>(slot));
}
}

// Base class handler implementation
QGraphicsLayoutItem* QGraphicsLinearLayout_QBaseItemAt(const QGraphicsLinearLayout* self, int index) {
	auto* vqgraphicslinearlayout = dynamic_cast<const VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_ItemAt_IsBase(true);
	return vqgraphicslinearlayout->itemAt(index);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLinearLayout_OnItemAt(const QGraphicsLinearLayout* self, intptr_t slot) {
	auto* vqgraphicslinearlayout = dynamic_cast<const VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_ItemAt_Callback(reinterpret_cast<VirtualQGraphicsLinearLayout::QGraphicsLinearLayout_ItemAt_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsLinearLayout_QBaseInvalidate(QGraphicsLinearLayout* self) {
	auto* vqgraphicslinearlayout = dynamic_cast<VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_Invalidate_IsBase(true);
	vqgraphicslinearlayout->invalidate();
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLinearLayout_OnInvalidate(QGraphicsLinearLayout* self, intptr_t slot) {
	auto* vqgraphicslinearlayout = dynamic_cast<VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_Invalidate_Callback(reinterpret_cast<VirtualQGraphicsLinearLayout::QGraphicsLinearLayout_Invalidate_Callback>(slot));
}
}

// Base class handler implementation
QSizeF* QGraphicsLinearLayout_QBaseSizeHint(const QGraphicsLinearLayout* self, int which, const QSizeF* constraint) {
	auto* vqgraphicslinearlayout = dynamic_cast<const VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_SizeHint_IsBase(true);
	return new QSizeF(vqgraphicslinearlayout->sizeHint(static_cast<Qt::SizeHint>(which), *constraint));
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLinearLayout_OnSizeHint(const QGraphicsLinearLayout* self, intptr_t slot) {
	auto* vqgraphicslinearlayout = dynamic_cast<const VirtualQGraphicsLinearLayout*>(self);
	if (vqgraphicslinearlayout && vqgraphicslinearlayout->isVirtualQGraphicsLinearLayout) {
vqgraphicslinearlayout->setQGraphicsLinearLayout_SizeHint_Callback(reinterpret_cast<VirtualQGraphicsLinearLayout::QGraphicsLinearLayout_SizeHint_Callback>(slot));
}
}

void QGraphicsLinearLayout_Delete(QGraphicsLinearLayout* self) {
    delete self;
}

